encoding/src/commonMain/kotlin/HexString.kt

```diff
@@ -0,0 +1,25 @@
+package com.juul.tuulbox.encoding
+
+private const val hexCode = "0123456789ABCDEF"
+
+/**
+ * Provides basic [ByteArray] to hex [String] conversion. Modified version of kotlinx.serialization's `HexConverter`:
+ * https://github.com/Kotlin/kotlinx.serialization/blob/43d5f7841fc744b072a636b712e194081456b5ba/formats/cbor/commonTest/src/kotlinx/serialization/HexConverter.kt
+ *
+ * For more robust (and actively maintained) solutions we may consider [Okio](https://github.com/square/okio) or
+ * [Krypto](https://github.com/korlibs/krypto).
+ */
+public fun ByteArray.toHexString(
+    separator: String? = " ",
+    prefix: String? = null,
+    lowerCase: Boolean = false
+): String {
+    val r = StringBuilder(size * 2)
```

Comment:
Because of the prefix and separator, this capacity is incorrect. Not a big deal since it will re-allocate, but since we _can_ get it right we might as well.

```suggestion
    if (size == 0) return ""
    val capacity = size * (2 + (prefix?.length ?: 0)) + (size - 1) * (separator?.length ?: 0)
    val r = StringBuilder(capacity)
```

Reply:
Fixed